pkg/controller/datavolume-controller_test.go

```diff
@@ -1563,6 +1563,204 @@ var _ = Describe("All DataVolume Tests", func() {
+			Expect(err).ToNot(HaveOccurred())
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(ErrInvalidTermMsg))
+			err = reconciler.client.Get(context.TODO(), types.NamespacedName{Name: "test-dv", Namespace: metav1.NamespaceDefault}, pod)
+			Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+			By("Checking error event recorded")
+			event := <-reconciler.recorder.(*record.FakeRecorder).Events
+			Expect(event).To(ContainSubstring("Size-detection pod failed due to"))
+		})
+
+		It("Should get the size from the size-detection pod", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			pvc := createPvcInStorageClass("test", metav1.NamespaceDefault, &scName, nil, nil, corev1.ClaimBound)
+			pvc.SetAnnotations(make(map[string]string))
+			pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+			reconciler := createDatavolumeReconciler(dv, pvc, storageProfile, sc)
+
+			// Prepare the size-detection Pod with the required information
+			pod := reconciler.makeSizeDetectionPodSpec(pvc, dv)
+			pod.Status.Phase = corev1.PodSucceeded
+			pod.Status.ContainerStatuses = []corev1.ContainerStatus{
+				{
+					State: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{
+							ExitCode: 0,
+							Message:  "100", // Mock value
+						},
+					},
+				},
+			}
+			err := reconciler.client.Create(context.TODO(), pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Get the expected value
+			pvcSpec, err := RenderPvcSpec(reconciler.client, reconciler.recorder, reconciler.log, dv)
+			Expect(err).ToNot(HaveOccurred())
+			expectedSize, err := inflateSizeWithOverhead(reconciler.client, int64(100), pvcSpec)
+
+			// Checks
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(dv.GetAnnotations()[AnnPermissiveClone]).To(Equal("true"))
+			targetSize := pvcSpec.Resources.Requests[corev1.ResourceStorage]
+			Expect(targetSize.Cmp(expectedSize)).To(Equal(0))
+		})
+
+		It("Should get the size from the source PVC's annotations", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			// Prepare the source PVC with the required annotations
+			pvc := createPvcInStorageClass("test", metav1.NamespaceDefault, &scName, nil, nil, corev1.ClaimBound)
+			pvc.SetAnnotations(make(map[string]string))
+			pvc.GetAnnotations()[AnnVirtualImageSize] = "100" // Mock value
```

Comment:
Are we sure the `AnnVirtualImageSize` is always up to date? Is this AnnVirtualImageSize invalidated on updates or maybe compared to some other sizes/versions during read?

Reply:
ok, I have found it, I can see you double check by comparing the original Capacity with Current Capacity

Reply:
Is there a test for that scenario? It is pretty interesting scenario.

Reply:
There isn't, though I agree that it's an interesting case. I add it :+1:

Reply:
Is it already added? I cannot find it

Reply:
@brybacki yeah, this one: https://github.com/kubevirt/containerized-data-importer/blob/19c4ba477b5260bf9c75b97a4f17ec55e3304215/tests/cloner_test.go#L899

Notice that I've mocked the behavior instead of resizing the source PVC (I hope that's enough)

Reply:
I was looking at unit tests ;)